From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Tue, 5 Aug 2025 09:45:20 +0000
Subject: [PATCH] Restore //chrome/browser/password_entry_edit for local
 password manager

Adapted from GrapheneOS Vanadium patch 0210-Restore-chrome-browser-password_entry_edit-for-local.patch for the Vandium browser port.

Partially revert "[passwords/android] Remove unused password checkup UI"

This partially reverts commit 6875d2ce472e7f2097617525feec45313602e225.
---
 chrome/android/BUILD.gn                       |   4 +
 chrome/android/java/res/values/dimens.xml     |   6 +
 .../settings/FragmentDependencyProvider.java  |   7 +
 chrome/browser/BUILD.gn                       |   1 +
 .../android/java/res/values/strings.xml       |   2 +
 chrome/browser/password_entry_edit/BUILD.gn   |  11 +
 .../browser/password_entry_edit/DIR_METADATA  |   7 +
 chrome/browser/password_entry_edit/OWNERS     |   3 +
 .../password_entry_edit/android/BUILD.gn      |  77 ++++
 .../android/credential_edit_bridge.cc         | 139 ++++++
 .../android/credential_edit_bridge.h          |  96 ++++
 .../android/internal/BUILD.gn                 |  98 +++++
 .../BlockedCredentialViewBinder.java          |  44 ++
 .../CredentialEditBridge.java                 | 143 ++++++
 .../CredentialEditControllerTest.java         | 410 ++++++++++++++++++
 .../CredentialEditCoordinator.java            | 142 ++++++
 .../CredentialEditMediator.java               | 328 ++++++++++++++
 .../CredentialEditProperties.java             |  48 ++
 .../CredentialEditUiFactory.java              |  52 +++
 .../CredentialEditViewBinder.java             |  52 +++
 .../CredentialEditViewTest.java               | 237 ++++++++++
 .../FederatedCredentialViewBinder.java        |  45 ++
 .../res/layout/blocked_credential_view.xml    |  18 +
 .../java/res/layout/credential_edit_view.xml  | 141 ++++++
 .../res/layout/federated_credential_view.xml  |  86 ++++
 .../android/java/res/layout/site_or_app.xml   |  27 ++
 .../menu/credential_edit_action_bar_menu.xml  |  25 ++
 .../android/java/res/values/dimens.xml        |  17 +
 .../BlockedCredentialFragmentView.java        |  56 +++
 .../CredentialEditFragmentView.java           | 226 ++++++++++
 .../CredentialEntryFragmentViewBase.java      | 135 ++++++
 .../FederatedCredentialFragmentView.java      | 116 +++++
 .../browser/password_manager/android/BUILD.gn |   5 +
 .../PasswordAccessReauthenticationHelper.java | 143 ++++++
 .../PasswordReauthenticationFragment.java     | 111 +++++
 .../settings/ReauthenticationManager.java     | 198 +++++++++
 .../PasswordReauthenticationFragmentTest.java | 121 ++++++
 .../settings/ReauthenticationManagerTest.java | 214 +++++++++
 .../strings/android_chrome_strings.grd        |  83 ++++
 .../IDS_LOCKSCREEN_DESCRIPTION_EDIT.png.sha1  |   1 +
 .../IDS_PASSWORDS_CHECK_TITLE.png.sha1        |   1 +
 ...ECK_DELETE_CREDENTIAL_DIALOG_BODY.png.sha1 |   1 +
 .../IDS_PASSWORD_EDIT_HINT.png.sha1           |   1 +
 ...SSWORD_ENTRY_COPY_SET_SCREEN_LOCK.png.sha1 |   1 +
 ..._DELETE_CREDENTIAL_DIALOG_CONFIRM.png.sha1 |   1 +
 ...IT_DELETE_CREDENTIAL_DIALOG_TITLE.png.sha1 |   1 +
 ...D_ENTRY_EDIT_DELETION_DIALOG_BODY.png.sha1 |   1 +
 ...TRY_EDIT_DUPLICATE_USERNAME_ERROR.png.sha1 |   1 +
 ...ENTRY_VIEWER_SHOW_STORED_PASSWORD.png.sha1 |   1 +
 ...SSWORD_ENTRY_VIEW_SET_SCREEN_LOCK.png.sha1 |   1 +
 .../IDS_PASSWORD_VIA_FEDERATION.png.sha1      |   1 +
 51 files changed, 3686 insertions(+)
 create mode 100644 chrome/browser/password_entry_edit/BUILD.gn
 create mode 100644 chrome/browser/password_entry_edit/DIR_METADATA
 create mode 100644 chrome/browser/password_entry_edit/OWNERS
 create mode 100644 chrome/browser/password_entry_edit/android/BUILD.gn
 create mode 100644 chrome/browser/password_entry_edit/android/credential_edit_bridge.cc
 create mode 100644 chrome/browser/password_entry_edit/android/credential_edit_bridge.h
 create mode 100644 chrome/browser/password_entry_edit/android/internal/BUILD.gn
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/BlockedCredentialViewBinder.java
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEditBridge.java
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEditControllerTest.java
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEditCoordinator.java
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEditMediator.java
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEditProperties.java
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEditUiFactory.java
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEditViewBinder.java
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEditViewTest.java
 create mode 100644 chrome/browser/password_entry_edit/android/internal/java/src/org/chromium/chrome/browser/password_entry_edit/FederatedCredentialViewBinder.java
 create mode 100644 chrome/browser/password_entry_edit/android/java/res/layout/blocked_credential_view.xml
 create mode 100644 chrome/browser/password_entry_edit/android/java/res/layout/credential_edit_view.xml
 create mode 100644 chrome/browser/password_entry_edit/android/java/res/layout/federated_credential_view.xml
 create mode 100644 chrome/browser/password_entry_edit/android/java/res/layout/site_or_app.xml
 create mode 100644 chrome/browser/password_entry_edit/android/java/res/menu/credential_edit_action_bar_menu.xml
 create mode 100644 chrome/browser/password_entry_edit/android/java/res/values/dimens.xml
 create mode 100644 chrome/browser/password_entry_edit/android/java/src/org/chromium/chrome/browser/password_entry_edit/BlockedCredentialFragmentView.java
 create mode 100644 chrome/browser/password_entry_edit/android/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEditFragmentView.java
 create mode 100644 chrome/browser/password_entry_edit/android/java/src/org/chromium/chrome/browser/password_entry_edit/CredentialEntryFragmentViewBase.java
 create mode 100644 chrome/browser/password_entry_edit/android/java/src/org/chromium/chrome/browser/password_entry_edit/FederatedCredentialFragmentView.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordAccessReauthenticationHelper.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordReauthenticationFragment.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/ReauthenticationManager.java
 create mode 100644 chrome/browser/password_manager/android/junit/src/org/chromium/chrome/browser/password_manager/settings/PasswordReauthenticationFragmentTest.java
 create mode 100644 chrome/browser/password_manager/android/junit/src/org/chromium/chrome/browser/password_manager/settings/ReauthenticationManagerTest.java
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_LOCKSCREEN_DESCRIPTION_EDIT.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORDS_CHECK_TITLE.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_CHECK_DELETE_CREDENTIAL_DIALOG_BODY.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_EDIT_HINT.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_ENTRY_COPY_SET_SCREEN_LOCK.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_ENTRY_EDIT_DELETE_CREDENTIAL_DIALOG_CONFIRM.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_ENTRY_EDIT_DELETE_CREDENTIAL_DIALOG_TITLE.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_ENTRY_EDIT_DELETION_DIALOG_BODY.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_ENTRY_EDIT_DUPLICATE_USERNAME_ERROR.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_ENTRY_VIEWER_SHOW_STORED_PASSWORD.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_ENTRY_VIEW_SET_SCREEN_LOCK.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_VIA_FEDERATION.png.sha1

diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 32fe81226653a..f75134432adc4 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -3437,6 +3437,7 @@ static_library("browser") {
       "//chrome/browser/partnerbookmarks:android_impl",
       "//chrome/browser/partnercustomizations:jni_headers",
       "//chrome/browser/password_edit_dialog/android",
+      "//chrome/browser/password_entry_edit/android",
       "//chrome/browser/password_manager/android:password_manager_android",
       "//chrome/browser/policy/android",
       "//chrome/browser/policy/android:jni_headers",
diff --git a/chrome/browser/password_entry_edit/BUILD.gn b/chrome/browser/password_entry_edit/BUILD.gn
new file mode 100644
index 0000000000000..0fc6129fb9920
--- /dev/null
+++ b/chrome/browser/password_entry_edit/BUILD.gn
@@ -0,0 +1,11 @@
+# Copyright 2021 The Chromium Authors
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//build/config/android/rules.gni")
+java_group("public_java") {
+  deps = [
+    "android:factory_java",
+    "android:java",
+  ]
+}
diff --git a/chrome/browser/password_entry_edit/DIR_METADATA b/chrome/browser/password_entry_edit/DIR_METADATA
new file mode 100644
index 0000000000000..eed8b5f2b4731
--- /dev/null
+++ b/chrome/browser/password_entry_edit/DIR_METADATA
@@ -0,0 +1,7 @@
+monorail: {
+  component: "UI>Browser>Passwords"
+}
+team_email: "chromium-dev@chromium.org"
+buganizer_public: {
+  component_id: 1457410
+}
diff --git a/chrome/browser/password_entry_edit/OWNERS b/chrome/browser/password_entry_edit/OWNERS
new file mode 100644
index 0000000000000..b29db19810b4e
--- /dev/null
+++ b/chrome/browser/password_entry_edit/OWNERS
@@ -0,0 +1,3 @@
+friedrichh@chromium.org
+ioanap@chromium.org
+
